x-pack/plugin/esql/src/main/generated-src/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/InBooleanEvaluator.java

```diff
@@ -0,0 +1,173 @@
+            try (Releasable rhsRelease = Releasables.wrap(rhsBlocks)) {
+                for (int i = 0; i < rhsBlocks.length; i++) {
+                    rhsBlocks[i] = (BooleanBlock) rhs[i].eval(page);
+                }
+                BooleanVector lhsVector = lhsBlock.asVector();
+                if (lhsVector == null) {
+                    return eval(page.getPositionCount(), lhsBlock, rhsBlocks);
+                }
+                BooleanVector[] rhsVectors = new BooleanVector[rhs.length];
+                for (int i = 0; i < rhsBlocks.length; i++) {
+                    rhsVectors[i] = rhsBlocks[i].asVector();
+                    if (rhsVectors[i] == null) {
+                        return eval(page.getPositionCount(), lhsBlock, rhsBlocks);
+                    }
+                }
+                return eval(page.getPositionCount(), lhsVector, rhsVectors);
+            }
+        }
+    }
+
+    public BooleanBlock eval(int positionCount, BooleanBlock lhsBlock, BooleanBlock[] rhsBlocks) {
+        try (BooleanBlock.Builder result = driverContext.blockFactory().newBooleanBlockBuilder(positionCount)) {
+            boolean[] rhsValues = new boolean[rhs.length];
+            BitSet nulls = new BitSet(rhs.length);
+            BitSet mvs = new BitSet(rhs.length);
+            for (int p = 0; p < positionCount; p++) {
+                if (lhsBlock.isNull(p)) {
+                    result.appendNull();
+                    continue;
+                }
+                if (lhsBlock.getValueCount(p) != 1) {
+                    if (lhsBlock.getValueCount(p) > 1) {
+                        warnings.registerException(new IllegalArgumentException("single-value function encountered multi-value"));
+                    }
+                    result.appendNull();
+                    continue;
+                }
+                // unpack rhsBlocks into rhsValues
+                nulls.clear();
+                mvs.clear();
+                for (int i = 0; i < rhsBlocks.length; i++) {
+                    if (rhsBlocks[i].isNull(p)) {
+                        nulls.set(i);
+                        continue;
+                    }
+                    if (rhsBlocks[i].getValueCount(p) > 1) {
+                        mvs.set(i);
+                        warnings.registerException(new IllegalArgumentException("single-value function encountered multi-value"));
+                        continue;
+                    }
+                    int o = rhsBlocks[i].getFirstValueIndex(p);
+                    rhsValues[i] = rhsBlocks[i].getBoolean(o);
+                }
+                if (nulls.cardinality() == rhsBlocks.length || mvs.cardinality() == rhsBlocks.length) {
+                    result.appendNull();
+                    continue;
+                }
+                try {
+                    In.process(result, nulls, mvs, lhsBlock.getBoolean(lhsBlock.getFirstValueIndex(p)), rhsValues);
```

Comment:
I wonder - should `boolean`s be special and not use the array method? Ultimately `IN (true)` and `IN (false)` and `IN ()` and `IN (false, true)` are the only possible layouts. Right? `IN (false, false) has the same meaning as `IN (false)`. I think.

If that's true, then this is useful: I've found sometimes I don't use the string template stuff for `boolean` because it's so different. In this case maybe I'd slurp more of this method into `In.process`'s boolean flavor. But whatever make the hacking easy I'm for. I don't have a strong opinion about the right way to lay out these templates.

Reply:
Indeed, I'll simplify boolean.